### PR TITLE
Add versions by project endpoint

### PR DESCRIPTION
**Pull request by dbradf**

### CODE REVIEW

File: src/evergreen/api.py [hunk 1]
```diff
@@ -50,6 +51,8 @@
 START_WAIT_TIME_SEC = 2
 MAX_WAIT_TIME_SEC = 5
 
+Requester = Enum('Requester', 'patch_request gitter_request github_pull_request merge_test ad_hoc')
```
Comment (ancostas) [inline src/evergreen/api.py]: [nit]/[question] Are we using lowercase across the board? I think uppercase is more "standard" (signifying constant value, parity with other languages), and we could call `requester.name.lower()` below.

Perhaps worth making a dictionary keyed by values of this enum? In general we don't know that a web API will accept parameters that are conveniently similar to python syntax, and we would need to do (e.g. when API uses camelCase):
```
OPTIONS = {
    Requester.PATCH_REQUEST: 'patchRequest'
}
...
params = {
    'requester': OPTIONS[requester]
}
```

File: src/evergreen/cli/main.py [hunk 2]
```diff
@@ -75,6 +75,24 @@ def list_projects(ctx):
...
+    api = ctx.obj['api']
+    fmt = ctx.obj['format']
+    version_list = api.versions_by_project(project)
+    versions_to_display = []
+    for i, version in enumerate(version_list):
+        if i > limit:
+            break
+
+        versions_to_display.append(version)
+
```
Comment (ancostas) [inline src/evergreen/cli/main.py]: What do you think of `itertools.takewhile(lambda i, version: i <= limit, enumerate(version_list))` for this?

Comment (dbradf) [reply src/evergreen/cli/main.py]: updated.

File: src/evergreen/api.py [hunk 3]
```diff
@@ -146,6 +149,31 @@ def _paginate(self, url, params=None):
...
+        """
+        if not params:
+            params = {
+                'limit': DEFAULT_LIMIT,
+            }
+
+        next_url = url
+        while True:
+            response = self._call_api(next_url, params)
+            if not response.json():
```
[the remark this answers is not in the document]
Comment (dbradf) [reply src/evergreen/api.py]: updated.

File: src/evergreen/api.py (hunk 1 above)
Comment (dbradf) [reply src/evergreen/api.py]: updated.

**Pull request merged**